katas/arrdiff.py

```diff
@@ -1,11 +1,10 @@
-"""This is called arrdiff just for consistency with the code wars problem. I understand that I am actually using lists."""
+"""This is called arrdiff just for consistency with the code wars problem."""
 
 
-def array_diff(a, b):
+def array_diff(lst_a, lst_b):
     """returns list a without any of the numbers that are in list b."""
-    to_pop = []
-    for num_b in b:
-        for ind_a in range(len(a) - 1, -1, -1):
-            if a[ind_a] == num_b:
-                a.pop(ind_a)
-    return a
\ No newline at end of file
+    for num_b in lst_b:
+        for ind_a in range(len(lst_a) - 1, -1, -1):
+            if lst_a[ind_a] == num_b:
+                lst_a.pop(ind_a)
```

Comment:
This is a bad practice in Python (probably other languages too). You should not modify a list while iterating over it or its indices.

A better approach would be to construct a new list, e.g.
`return [num for num in lst_a if num not in lst_b]`